### PR TITLE
[Lib] Add support for Luis sentiment

### PR DESCRIPTION
**Pull request by DingmaomaoBJTU**
### Purpose
Add an extension to extract sentiment information from Luis recognizer result
#2623

### Changes
1. Add an extension to extract sentiment
2. Add related UT

Here is an example of how to use sentiment in skill:
(var type, var score) = emailLuisResult.GetSentimentInfo(li => li.Properties);

Before trying this feature, please ensure you have luis sentiment setting on:
image

BTW, the luis model published with sentiment setting on is blocked by:
https://github.com/microsoft/botbuilder-tools/issues/1340 
Will update luis deploy script once this issue is resolved.

### Tests
Yes

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

#### General
- [x] I have commented my code, particularly in hard-to-understand areas	
- [x] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

#### Bots
- [ ] I have validated that new and updated responses use appropriate [Speak](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-text-to-speech?view=azure-bot-service-3.0) and [InputHint](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-add-input-hints?view=azure-bot-service-3.0) properties to ensure a high-quality speech-first experience
- [ ] I have replicated language model changes across the English, French, Italian, German, Spanish, and Chinese `.lu` files and validated that deployment is successful

#### Deployment Scripts
- [ ] I have replicated my changes in the **Virtual Assistant Template** and **Sample** projects
- [ ] I have replicated my changes in the **Skill Template** and **Sample** projects

### CODE REVIEW

File: sdk/csharp/libraries/microsoft.bot.builder.solutions/Microsoft.Bot.Builder.Solutions.csproj [hunk 1]
```diff
@@ -15,6 +15,7 @@
     <Company>Microsoft</Company>
     <Authors>Microsoft</Authors>
     <Description>Shared library for Conversational AI Virtual Assistants and Skills.</Description>
+    <LangVersion>7.1</LangVersion>
```
Comment (lzc850612) [inline sdk/csharp/libraries/microsoft.bot.builder.solutions/Microsoft.Bot.Builder.Solutions.csproj]: is this required?

**Review by lzc850612 — CHANGES_REQUESTED** [review]

File: sdk/csharp/libraries/microsoft.bot.builder.solutions/Microsoft.Bot.Builder.Solutions.csproj (hunk 1 above)
Comment (DingmaomaoBJTU) [reply sdk/csharp/libraries/microsoft.bot.builder.solutions/Microsoft.Bot.Builder.Solutions.csproj]: Removed~ This is to fix a warning in VS2017. When change to VS2019, the error is gone.

**Review by lzc850612 — APPROVED** [review]

File: sdk/csharp/libraries/microsoft.bot.builder.solutions/Extensions/LuisRecognizerEx.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            {
+                var sentimentInfo = JsonConvert.DeserializeObject<Sentiment>(result.ToString());
+                sentimentLabel = GetSentimentType(sentimentInfo.Label);
+                maxScore = sentimentInfo.Score.HasValue ? sentimentInfo.Score.Value : 0.0;
+            }
+
+            return (sentimentLabel, maxScore);
+        }
+
+        public static SentimentType GetSentimentType(string label)
```
Comment (lauren-mills) [inline sdk/csharp/libraries/microsoft.bot.builder.solutions/Extensions/LuisRecognizerEx.cs:33]: For this, have you checked if the Microsoft.Azure.CognitiveServices.Language.TextAnalytics library has any classes we could resuse for this sentiment result? I believe that's what LUIS is using under the covers so would be good to align if possible.

Comment (DingmaomaoBJTU) [reply sdk/csharp/libraries/microsoft.bot.builder.solutions/Extensions/LuisRecognizerEx.cs:33]: Yes, it use https://github.com/Azure/azure-sdk-for-net/blob/8a5ac017ff3bfd11988bb9ed6d566606c166d0ea/sdk/cognitiveservices/Language.LUIS.Runtime/src/Generated/Models/Sentiment.cs this class to store sentiment result. But one problem is the sentiment type it has is a string, which is not convenience to use, so I add a enum type~

**Comment by DingmaomaoBJTU** [comment]
@lauren-mills could you help take a look?

**Review by lauren-mills — APPROVED** [review]

**Pull request merged**